app/lib/report_helper.rb

```diff
@@ -530,20 +530,22 @@ def self.fetch_top_taxons(samples, background_id, categories, read_specificity =
         pr = pipeline_runs_by_id[pipeline_run_id]
         result_hash[pipeline_run_id] = { "pr" => pr, "taxon_counts" => [] }
       end
-      row["rpm"] = row["r"] / ((pr.total_reads - pr.total_ercc_reads.to_i) * pr.subsample_fraction) * 1_000_000.0
-      row["zscore"] = row["stdev"].nil? ? ZSCORE_WHEN_ABSENT_FROM_BACKGROUND : ((row["rpm"] - row["mean"]) / row["stdev"])
-      row["zscore"] = ZSCORE_MAX if row["zscore"] > ZSCORE_MAX && row["zscore"] != ZSCORE_WHEN_ABSENT_FROM_BACKGROUND
-      row["zcore"] = ZSCORE_MIN if row["zscore"] < ZSCORE_MIN
-      result_hash[pipeline_run_id]["taxon_counts"] << row
+      if pr.total_reads
+        row["rpm"] = row["r"] / ((pr.total_reads - pr.total_ercc_reads.to_i) * pr.subsample_fraction) * 1_000_000.0
+        row["zscore"] = row["stdev"].nil? ? ZSCORE_WHEN_ABSENT_FROM_BACKGROUND : ((row["rpm"] - row["mean"]) / row["stdev"])
+        row["zscore"] = ZSCORE_MAX if row["zscore"] > ZSCORE_MAX && row["zscore"] != ZSCORE_WHEN_ABSENT_FROM_BACKGROUND
+        row["zcore"] = ZSCORE_MIN if row["zscore"] < ZSCORE_MIN
```

Comment:
Should this be `zscore` instead of `zcore`? I realize this was there before.

Reply:
Nice catch! It looks like a bug. The effect would have been to not truncate low value z-scores.

Reply:
Maybe this is a good time to write a unit test that verifies the numbers stay the same, just like the report has :)

Reply:
Oh and I see that this was actually a pre-existing bug.